### PR TITLE
Update decoder of s210x and s2120

### PR DESCRIPTION
**Pull request by Kway99**
I have updated the driver of s210x and s2120 based on your comments. Please kindly have a check! much appreciated!

### CODE REVIEW

**Comment by mostafaibr** [comment]
@Kway99 Thank you for the modifications. 
However the unit tests do not pass on your drivers. Please run `npm install && npm run test` to see the results and diff. 

For s210x: 
```
 FAIL  ./driver-examples.spec.js
  Decode uplink
    ✕ uplink containing only pulse counter (8 ms)

  ● Decode uplink › uplink containing only pulse counter

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 21
    + Received  +  6

      Object {
        "data": Object {
          "0": Object {
    -       "measurementId": "4097",
    -       "measurementValue": 27.2,
    -       "type": "Air Temperature",
    +       "measurementId": 3,
    +       "measurementValue": 825307.237,
    +       "type": "Measurement",
          },
          "1": Object {
    -       "measurementId": "4098",
    -       "measurementValue": "-24",
    -       "type": "Air Humidity",
    -     },
    -     "2": Object {
    -       "measurementId": "4099",
    -       "measurementValue": "-1758103807",
    -       "type": "Light Intensity",
    -     },
    -     "3": Object {
    -       "measurementId": "4190",
    -       "measurementValue": 1,
    -       "type": "UV Index",
    -     },
    -     "4": Object {
    -       "measurementId": "4105",
    -       "measurementValue": 432.8,
    -       "type": "Wind Speed",
    +       "measurementId": 1,
    +       "measurementValue": 943273.779,
    +       "type": "Measurement",
          },
        },
        "errors": Array [],
        "warnings": Array [],
      }

      24 |         // Then
      25 |         const expected = example.output
    > 26 |         expect(result).toStrictEqual(expected)
         |                        ^
      27 |       })
      28 |     }
      29 |   })

      at Object.toStrictEqual (driver-examples.spec.js:26:24)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.63 s
```

For s2120: 
``` 
 FAIL  ./driver-examples.spec.js
  Decode uplink
    ✕ uplink containing only pulse counter (8 ms)

  ● Decode uplink › uplink containing only pulse counter

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 21
    + Received  +  6

      Object {
        "data": Object {
          "0": Object {
    -       "measurementId": "4097",
    -       "measurementValue": 27.2,
    -       "type": "Air Temperature",
    +       "measurementId": 3,
    +       "measurementValue": 825307.237,
    +       "type": "Measurement",
          },
          "1": Object {
    -       "measurementId": "4098",
    -       "measurementValue": "-24",
    -       "type": "Air Humidity",
    -     },
    -     "2": Object {
    -       "measurementId": "4099",
    -       "measurementValue": "-1758103807",
    -       "type": "Light Intensity",
    -     },
    -     "3": Object {
    -       "measurementId": "4190",
    -       "measurementValue": 1,
    -       "type": "UV Index",
    -     },
    -     "4": Object {
    -       "measurementId": "4105",
    -       "measurementValue": 432.8,
    -       "type": "Wind Speed",
    +       "measurementId": 1,
    +       "measurementValue": 943273.779,
    +       "type": "Measurement",
          },
        },
        "errors": Array [],
        "warnings": Array [],
      }

      24 |         // Then
      25 |         const expected = example.output
    > 26 |         expect(result).toStrictEqual(expected)
         |                        ^
      27 |       })
      28 |     }
      29 |   })

      at Object.toStrictEqual (driver-examples.spec.js:26:24)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.63 s
```

File: vendors/sensecap/drivers/s2120/index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ function decodeUplink(input) {
     }
 
     var bytes = input['bytes']
-    bytes = Buffer.from(bytes, 'utf8').toString()
+    bytes = Buffer.from(bytes, 'utf8').toString('hex')
```
Comment (mostafaibr) [inline vendors/sensecap/drivers/s2120/index.js:11]:
```suggestion
    bytes = Buffer.from(bytes).toString('hex')
```

the input.bytes is an array of integers, thus, this conversion does not work. You should not specify the source type as 'utf8'.

**Review by mostafaibr — CHANGES_REQUESTED** [review]

**Comment by mostafaibr** [comment]
In the updated drivers, the version in package.json should be updated. 
Here the modifications are bugs fix, so kindly update the version from 1.0.0 to 1.0.1

**Comment by Kway99** [comment]
Hello @mostafaibr , Thanks for your suggestion! We have updated the driver files for S210X and S2120. And we also run the unit test locally. Check this:
S210x
企业微信截图_06c5fabf-a64a-4526-8ee1-d231a9b82b81
S2120
企业微信截图_6e8b36c8-2e36-42c7-be92-8006af0796ff

**Pull request merged**